File: API/Entities/Basket.cs
namespace API.Entities;

public class Basket
{
    public int Id { get; set; }
    public string ClientId { get; set; }
    public List<BasketItem> Items { get; set; } = new();
    public string PaymentIntentId { get; set; }
    public string ClientSecret { get; set; }

    public void AddItem(Product product, int quantity, int sizeMl, int pricePercent)
    {
        if (Items.All(item => item.ProductId != product.Id || item.SizeMl != sizeMl))
        {
            Items.Add(new BasketItem { Product = product, Quantity = quantity ,SizeMl = sizeMl, PricePercent = pricePercent });
        }

        var existingItem = Items.FirstOrDefault(item => item.ProductId == product.Id);
        if (existingItem != null) existingItem.Quantity += quantity;
    }

    public void RemoveItem(int productId, int quantity, int sizeMl)
    {
        var item = Items.FirstOrDefault(item => item.ProductId == productId && item.SizeMl == sizeMl);

        if (item == null) return;

        item.Quantity -= quantity;

        if (item.Quantity == 0) Items.Remove(item);
    }
}

File: client/src/app/models/basket.ts
export interface Basket {
    id: number;
    clientId: string;
    items: BasketItem[];
    paymentIntentId: string;
    clientSecret: string;
}

export interface BasketItem {
    productId: number;
    name: string;
    price: number;
    pricePercent: number;
    sizeMl: number;
    pictureUrl: string;
    brand: string;
    quantity: number;
}